Error when importing MongoDB database with collection having nested fields with more than 1 level of nesting

### Describe the bug

Using the serialization feature, importing a MongoDB database containing a collection with nested objects is failing.
Example of such collection:
```JSON
{"_id":{"$oid":"65d636516d51677dd86e8d04"},"level_0":{"level_1":{"level_2":"test"}}}
```
level_0 is a parent object having a level_1 child object containing a level_2 string

### To Reproduce

1. Create a MongoDB collection inside a database with the following document:
    ```JSON
    {"_id":{"$oid":"65d636516d51677dd86e8d04"},"level_0":{"level_1":{"level_2":"test"}}}
    ```
2. Launch a new instance of Metabase (I'm using the docker image) and add the connection to the MongoDB database containing that collection.
3. Export the Metabase database with
    ```SHELL
    java -jar /app/metabase.jar export exportFolder --no-settings
    ```
4. Copy the export folder exportFolder to your local drive
5. Remove the container and start a new fresh Metabase instance/container.
6. Copy the export folder exportFolder from your local drive to the Metabase instance
7. Import the Metabase database with
   ```SHELL
    java -jar /app/metabase.jar import exportFolder
    ```

### Expected behavior

The import should succeed without errors, if any cards/questions are built upon this MongoDB collection, they should also be working fine.

### Logs

<details>
<summary>Click to expand the logs from the import command:</summary>

```
Warning: environ value jdk-11.0.22+7 for key :java-version has been overwritten with 11.0.22
2024-03-25 09:05:00,174 INFO metabase.util :: Maximum memory available to JVM: 3.9 GB
2024-03-25 09:05:01,729 WARN db.env :: WARNING: Using Metabase with an H2 application database is not recommended for production deployments. For production deployments, we highly recommend using Postgres, MySQL, or MariaDB instead. If you decide to continue to use H2, please be sure to back up the database file regularly. For more information, see https://metabase.com/docs/latest/operations-guide/migrating-from-h2.html
2024-03-25 09:05:01,870 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓 
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
2024-03-25 09:05:03,273 ERROR public-settings.premium-features :: Error validating token : Assert failed: Metabase DB is not yet set up
((requiring-resolve (quote metabase.db/db-is-set-up?)))
2024-03-25 09:05:06,684 INFO driver.impl :: Registered abstract driver :sql  🚚
2024-03-25 09:05:06,694 INFO driver.impl :: Registered abstract driver :sql-jdbc (parents: [:sql]) 🚚
2024-03-25 09:05:06,698 INFO metabase.util :: Load driver :sql-jdbc took 73.6 ms
2024-03-25 09:05:06,699 INFO driver.impl :: Registered driver :h2 (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:06,831 INFO driver.impl :: Registered driver :mysql (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:06,858 INFO driver.impl :: Registered driver :postgres (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:09,587 INFO metabase.core :: 
Metabase v1.48.4 (62145b0)

Copyright © 2024 Metabase, Inc.

Metabase Enterprise Edition extensions are PRESENT.

Usage of Metabase Enterprise Edition features are subject to the Metabase Commercial License. See https://www.metabase.com/license/commercial/ for details.
2024-03-25 09:05:09,731 INFO metabase.plugins :: Loading plugins in /home/metabase/plugins...
2024-03-25 09:05:10,098 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :bigquery-cloud-sdk...
2024-03-25 09:05:10,106 INFO driver.impl :: Registered driver :bigquery-cloud-sdk (parents: [:sql]) 🚚
2024-03-25 09:05:10,111 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :sqlite...
2024-03-25 09:05:10,112 INFO driver.impl :: Registered driver :sqlite (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,123 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :mongo...
2024-03-25 09:05:10,124 INFO driver.impl :: Registered driver :mongo  🚚
2024-03-25 09:05:10,130 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :googleanalytics...
2024-03-25 09:05:10,130 INFO driver.impl :: Registered driver :googleanalytics  🚚
2024-03-25 09:05:10,136 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :vertica...
2024-03-25 09:05:10,137 INFO driver.impl :: Registered driver :vertica (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,148 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :athena...
2024-03-25 09:05:10,149 INFO driver.impl :: Registered driver :athena (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,154 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :sqlserver...
2024-03-25 09:05:10,155 INFO driver.impl :: Registered driver :sqlserver (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,179 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :snowflake...
2024-03-25 09:05:10,180 INFO driver.impl :: Registered driver :snowflake (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,188 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :oracle...
2024-03-25 09:05:10,189 INFO driver.impl :: Registered driver :oracle (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,193 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :redshift...
2024-03-25 09:05:10,194 INFO driver.impl :: Registered driver :redshift (parents: [:postgres]) 🚚
2024-03-25 09:05:10,200 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :hive-like...
2024-03-25 09:05:10,201 INFO driver.impl :: Registered abstract driver :hive-like (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,202 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :sparksql...
2024-03-25 09:05:10,203 INFO driver.impl :: Registered driver :sparksql (parents: [:hive-like]) 🚚
2024-03-25 09:05:10,214 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :presto-jdbc...
2024-03-25 09:05:10,215 INFO driver.impl :: Registered driver :presto-jdbc (parents: [:sql-jdbc]) 🚚
2024-03-25 09:05:10,220 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :druid...
2024-03-25 09:05:10,220 INFO driver.impl :: Registered driver :druid  🚚
2024-03-25 09:05:10,223 INFO db.setup :: Verifying h2 Database Connection ...
2024-03-25 09:05:10,712 INFO db.setup :: Successfully verified H2 2.1.214 (2022-06-13) application database connection. ✅
2024-03-25 09:05:10,714 INFO db.setup :: Checking if a database downgrade is required...
2024-03-25 09:05:11,073 INFO db.setup :: Running Database Migrations...
2024-03-25 09:05:11,074 INFO db.setup :: Setting up Liquibase...
2024-03-25 09:05:11,119 INFO db.setup :: Liquibase is ready.
2024-03-25 09:05:11,120 INFO db.liquibase :: Checking if Database has unrun migrations...
2024-03-25 09:05:11,538 INFO db.liquibase :: No unrun migrations found.
2024-03-25 09:05:11,539 INFO db.setup :: Database Migrations Current ...  ✅
2024-03-25 09:05:11,541 INFO metabase.util :: Database setup took 1.3 s
2024-03-25 09:05:11,625 INFO public-settings.premium-features :: GETTING ACTIVE USER COUNT!
2024-03-25 09:05:11,666 INFO public-settings.premium-features :: => 1
2024-03-25 09:05:11,666 INFO public-settings.premium-features :: Checking with the MetaStore to see whether token '82a5...30d4' is valid...
2024-03-25 09:05:12,353 INFO serialization.cmd :: Loading serialized Metabase files from export
2024-03-25 09:05:12,453 INFO v2.load :: Loading Database sample-db
2024-03-25 09:05:12,557 INFO driver.impl :: Initializing driver :sql...
2024-03-25 09:05:12,558 INFO driver.impl :: Initializing driver :sql-jdbc...
2024-03-25 09:05:12,559 INFO driver.impl :: Initializing driver :h2...
2024-03-25 09:05:12,607 INFO driver.impl :: Initializing driver :mongo...
2024-03-25 09:05:12,609 INFO plugins.classloader :: Added URL file:/home/metabase/plugins/mongo.metabase-driver.jar to classpath
2024-03-25 09:05:12,611 DEBUG plugins.init-steps :: Loading plugin namespace metabase.driver.mongo...
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: monger.util, being replaced by: #'monger.util/random-uuid
2024-03-25 09:05:13,290 INFO driver.impl :: Registered driver :mongo  🚚
2024-03-25 09:05:13,306 INFO metabase.util :: Load lazy loading driver :mongo took 695.7 ms
2024-03-25 09:05:13,436 INFO task.sync-databases :: Scheduling sync/analyze for database 2: trigger: metabase.task.sync-and-analyze.trigger.2
2024-03-25 09:05:13,438 INFO task.sync-databases :: Scheduling field-values for database 2: trigger: metabase.task.update-field-values.trigger.2
2024-03-25 09:05:13,440 INFO v2.load :: Loading Database sample-db > Table gs_7334
2024-03-25 09:05:13,443 INFO v2.load :: Loading Database sample-db
2024-03-25 09:05:13,504 INFO v2.load :: Loading Database sample-db > Table gs_7334 > Field level_2
2024-03-25 09:05:13,506 INFO v2.load :: Loading Database sample-db > Table gs_7334
clojure.lang.ExceptionInfo: Failed to load into database for [#ordered/map ([:id "sample-db"] [:model "Database"]) #ordered/map ([:id "gs_7334"] [:model "Table"]) #ordered/map ([:id "level_2"] [:model "Field"])] {:path [#ordered/map ([:id "sample-db"] [:model "Database"]) #ordered/map ([:id "gs_7334"] [:model "Table"]) #ordered/map ([:id "level_2"] [:model "Field"])], :deps-chain #{}}
        at metabase_enterprise.serialization.v2.load$load_one_BANG_.invokeStatic(load.clj:70)
        at metabase_enterprise.serialization.v2.load$load_one_BANG_.invoke(load.clj:33)
        at clojure.core.protocols$fn__8249.invokeStatic(protocols.clj:168)
        at clojure.core.protocols$fn__8249.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at metabase_enterprise.serialization.v2.load$load_metabase_BANG_.invokeStatic(load.clj:98)
        at metabase_enterprise.serialization.v2.load$load_metabase_BANG_.doInvoke(load.clj:83)
        at clojure.lang.RestFn.invoke(RestFn.java:423)
        at metabase_enterprise.serialization.cmd$fn__108901$_AMPERSAND_f__108903.doInvoke(cmd.clj:100)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:671)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase_enterprise.serialization.cmd$fn__108901$fn__108908.doInvoke(cmd.clj:80)
        at clojure.lang.RestFn.invoke(RestFn.java:464)
        at metabase_enterprise.serialization.cmd$fn__108911$_AMPERSAND_f__108912.invoke(cmd.clj:110)
        at metabase_enterprise.serialization.cmd$fn__108911$fn__108914.invoke(cmd.clj:102)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase.cmd$call_enterprise.invokeStatic(cmd.clj:54)
        at metabase.cmd$call_enterprise.doInvoke(cmd.clj:43)
        at clojure.lang.RestFn.invoke(RestFn.java:439)
        at metabase.cmd$import.invokeStatic(cmd.clj:192)
        at metabase.cmd$import.doInvoke(cmd.clj:188)
        at clojure.lang.RestFn.invoke(RestFn.java:410)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase.cmd$run_cmd$fn__107980.invoke(cmd.clj:301)
        at metabase.cmd$run_cmd.invokeStatic(cmd.clj:300)
        at metabase.cmd$run_cmd.invoke(cmd.clj:290)
        at clojure.lang.Var.invoke(Var.java:388)
        at metabase.core$run_cmd.invokeStatic(core.clj:178)
        at metabase.core$run_cmd.invoke(core.clj:176)
        at metabase.core$entrypoint.invokeStatic(core.clj:200)
        at metabase.core$entrypoint.doInvoke(core.clj:195)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
        at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at metabase.bootstrap.main(Unknown Source)
Caused by: clojure.lang.ExceptionInfo: Referential integrity constraint violation: "FK_FIELD_PARENT_REF_FIELD_ID: PUBLIC.METABASE_FIELD FOREIGN KEY(PARENT_ID) REFERENCES PUBLIC.METABASE_FIELD(ID) (249)"; SQL statement:
INSERT INTO "METABASE_FIELD" ("DESCRIPTION", "DATABASE_TYPE", "SEMANTIC_TYPE", "TABLE_ID", "COERCION_STRATEGY", "NAME", "HAS_FIELD_VALUES", "SETTINGS", "CAVEATS", "FK_TARGET_FIELD_ID", "UPDATED_AT", "CUSTOM_POSITION", "EFFECTIVE
_TYPE", "ACTIVE", "NFC_PATH", "PARENT_ID", "DATABASE_IS_AUTO_INCREMENT", "JSON_UNFOLDING", "POSITION", "VISIBILITY_TYPE", "PREVIEW_DISPLAY", "DISPLAY_NAME", "DATABASE_POSITION", "DATABASE_REQUIRED", "CREATED_AT", "BASE_TYPE", "P
OINTS_OF_INTEREST") VALUES (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NULL, NULL, NOW(), ?, ?, TRUE, NULL, ?, FALSE, FALSE, ?, ?, TRUE, ?, ?, FALSE, ?, ?, NULL) [23506-214] {:toucan2/context-trace [["execute SQL with class com.mcha
nge.v2.c3p0.impl.NewProxyConnection" {:toucan2.jdbc.query/sql-args ["INSERT INTO \"METABASE_FIELD\" (\"DESCRIPTION\", \"DATABASE_TYPE\", \"SEMANTIC_TYPE\", \"TABLE_ID\", \"COERCION_STRATEGY\", \"NAME\", \"HAS_FIELD_VALUES\", \"S
ETTINGS\", \"CAVEATS\", \"FK_TARGET_FIELD_ID\", \"UPDATED_AT\", \"CUSTOM_POSITION\", \"EFFECTIVE_TYPE\", \"ACTIVE\", \"NFC_PATH\", \"PARENT_ID\", \"DATABASE_IS_AUTO_INCREMENT\", \"JSON_UNFOLDING\", \"POSITION\", \"VISIBILITY_TYP
E\", \"PREVIEW_DISPLAY\", \"DISPLAY_NAME\", \"DATABASE_POSITION\", \"DATABASE_REQUIRED\", \"CREATED_AT\", \"BASE_TYPE\", \"POINTS_OF_INTEREST\") VALUES (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NULL, NULL, NOW(), ?, ?, TRUE, NULL,
 ?, FALSE, FALSE, ?, ?, TRUE, ?, ?, FALSE, ?, ?, NULL)" "java.lang.String" 22 "level_2" 0 "type/Text" 249 3 "normal" "Level 2" 3 #t "2024-03-25T08:41:04.560274" "type/Text"]}] ["resolve connection" {:toucan2.connection/connectab
le org.h2.jdbc.JdbcConnection}] ["resolve connection" {:toucan2.connection/connectable nil}] {:toucan2.pipeline/rf #object[clojure.core$map$fn__5931$fn__5932 0x370f893e "clojure.core$map$fn__5931$fn__5932@370f893e"]} ["with comp
iled query" {:toucan2.pipeline/compiled-query ["INSERT INTO \"METABASE_FIELD\" (\"DESCRIPTION\", \"DATABASE_TYPE\", \"SEMANTIC_TYPE\", \"TABLE_ID\", \"COERCION_STRATEGY\", \"NAME\", \"HAS_FIELD_VALUES\", \"SETTINGS\", \"CAVEATS\
", \"FK_TARGET_FIELD_ID\", \"UPDATED_AT\", \"CUSTOM_POSITION\", \"EFFECTIVE_TYPE\", \"ACTIVE\", \"NFC_PATH\", \"PARENT_ID\", \"DATABASE_IS_AUTO_INCREMENT\", \"JSON_UNFOLDING\", \"POSITION\", \"VISIBILITY_TYPE\", \"PREVIEW_DISPLA
Y\", \"DISPLAY_NAME\", \"DATABASE_POSITION\", \"DATABASE_REQUIRED\", \"CREATED_AT\", \"BASE_TYPE\", \"POINTS_OF_INTEREST\") VALUES (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NULL, NULL, NOW(), ?, ?, TRUE, NULL, ?, FALSE, FALSE, ?, 
?, TRUE, ?, ?, FALSE, ?, ?, NULL)" "java.lang.String" 22 "level_2" 0 "type/Text" 249 3 "normal" "Level 2" 3 #t "2024-03-25T08:41:04.560274" "type/Text"]}] ["with built query" {:toucan2.pipeline/built-query {:insert-into [:metaba
se_field], :values ((toucan2.instance/instance :model/Field {:description nil, :database_type "java.lang.String", :semantic_type nil, :table_id 22, :coercion_strategy nil, :name "level_2", :has_field_values nil, :settings nil, :
caveats nil, :fk_target_field_id nil, :updated_at [:metabase.util.honey-sql-2/typed :%now {:metabase.util.honeysql-extensions/database-type "timestamp"}], :custom_position 0, :effective_type "type/Text", :active true, :nfc_path 
nil, :parent_id 249, :database_is_auto_increment false, :json_unfolding false, :position 3, :visibility_type "normal", :preview_display true, :display_name "Level 2", :database_position 3, :database_required false, :created_at #
t "2024-03-25T08:41:04.560274", :base_type "type/Text", :points_of_interest nil}))}}] ["resolve connection" {:toucan2.connection/connectable metabase.db.connection.ApplicationDB}] ["resolve connection" {:toucan2.connection/conne
ctable :default}] ["resolve connection" {:toucan2.connection/connectable nil}] ["with resolved query" {:toucan2.pipeline/resolved-query {}}] ["with parsed args" {:toucan2.pipeline/query-type :toucan.query-type/insert.instances, 
:toucan2.pipeline/parsed-args {:rows [{:description nil, :database_type "java.lang.String", :semantic_type nil, :table_id 22, :coercion_strategy nil, :name "level_2", :has_field_values nil, :settings nil, :caveats nil, :fk_targe
t_field_id nil, :custom_position 0, :effective_type "type/Text", :active true, :nfc_path nil, :parent_id 249, :database_is_auto_increment false, :json_unfolding false, :position 3, :visibility_type "normal", :preview_display tru
e, :display_name "Level 2", :database_position 3, :database_required false, :created_at #t "2024-03-25T08:41:04.560274", :base_type "type/Text", :points_of_interest nil}]}}] ["with model" {:toucan2.pipeline/model :model/Field}] 
["with unparsed args" {:toucan2.pipeline/query-type :toucan.query-type/insert.instances, :toucan2.pipeline/unparsed-args (:model/Field {:description nil, :database_type "java.lang.String", :semantic_type nil, :table_id 22, :coer
cion_strategy nil, :name "level_2", :has_field_values nil, :settings nil, :caveats nil, :fk_target_field_id nil, :custom_position 0, :effective_type "type/Text", :active true, :nfc_path nil, :parent_id 249, :database_is_auto_inc
rement false, :json_unfolding false, :position 3, :visibility_type "normal", :preview_display true, :display_name "Level 2", :database_position 3, :database_required false, :created_at #t "2024-03-25T08:41:04.560274", :base_type "type/Text", :points_of_interest nil})}]]}
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
        at org.h2.message.DbException.get(DbException.java:223)
        at org.h2.message.DbException.get(DbException.java:199)
        at org.h2.constraint.ConstraintReferential.checkRowOwnTable(ConstraintReferential.java:311)
        at org.h2.constraint.ConstraintReferential.checkRow(ConstraintReferential.java:252)
        at org.h2.table.Table.fireConstraints(Table.java:1172)
        at org.h2.table.Table.fireAfterRow(Table.java:1190)
        at org.h2.command.dml.Insert.insertRows(Insert.java:188)
        at org.h2.command.dml.Insert.update(Insert.java:135)
        at org.h2.command.CommandContainer.executeUpdateWithGeneratedKeys(CommandContainer.java:242)
        at org.h2.command.CommandContainer.update(CommandContainer.java:163)
        at org.h2.command.Command.executeUpdate(Command.java:252)
        at org.h2.jdbc.JdbcPreparedStatement.execute(JdbcPreparedStatement.java:254)
        at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.execute(NewProxyPreparedStatement.java:67)
        at toucan2.jdbc.query$reduce_jdbc_query.invokeStatic(query.clj:40)
        at toucan2.jdbc.query$reduce_jdbc_query.invoke(query.clj:22)
        at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invokeStatic(pipeline.clj:19)
        at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invoke(pipeline.clj:9)
        at clojure.lang.AFn.applyToHelper(AFn.java:178)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:482)
        at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan_result_type_pks_default.invokeStatic(pipeline.clj:30)
        at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan_result_type_pks_default.invoke(pipeline.clj:22)
        at clojure.lang.AFn.applyToHelper(AFn.java:178)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17911.invoke(threaded.clj:79)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.doInvoke(core.clj:2589)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$apply.doInvoke(core.clj:662)
        at clojure.lang.RestFn.invoke(RestFn.java:533)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.doInvoke(threaded.clj:46)
        at clojure.lang.RestFn.applyTo(RestFn.java:151)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$apply.doInvoke(core.clj:662)
        at clojure.lang.RestFn.invoke(RestFn.java:533)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
        at methodical.impl.standard$invoke_multifn.doInvoke(standard.clj:47)
        at clojure.lang.RestFn.invoke(RestFn.java:594)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:199)
        at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan2_jdbc_pipeline_DML_queries_returning_instances_default.invokeStatic(pipeline.clj:62)
        at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan2_jdbc_pipeline_DML_queries_returning_instances_default.invoke(pipeline.clj:52)
        at clojure.lang.AFn.applyToHelper(AFn.java:178)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17911.invoke(threaded.clj:79)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.doInvoke(core.clj:2589)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$apply.doInvoke(core.clj:662)
        at clojure.lang.RestFn.invoke(RestFn.java:533)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.doInvoke(threaded.clj:46)
        at clojure.lang.RestFn.applyTo(RestFn.java:151)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$apply.doInvoke(core.clj:662)
        at clojure.lang.RestFn.invoke(RestFn.java:533)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
        at methodical.impl.standard$invoke_multifn.doInvoke(standard.clj:47)
        at clojure.lang.RestFn.invoke(RestFn.java:594)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:199)
        at toucan2.pipeline$transduce_execute$with_connection_STAR___21132$with_transaction_STAR___21133.invoke(pipeline.clj:75)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:190)
        at metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:172)
        at clojure.lang.AFn.applyToHelper(AFn.java:165)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:457)
        at clojure.core$partial$fn__5908.invoke(core.clj:2643)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17907.invoke(threaded.clj:71)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2588)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:44)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)
        at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)
        at clojure.lang.AFn.applyToHelper(AFn.java:165)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:457)
        at clojure.core$partial$fn__5908.invoke(core.clj:2643)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:195)
        at toucan2.pipeline$transduce_execute$with_connection_STAR___21132.invoke(pipeline.clj:74)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invokeStatic(connection.clj:13)
        at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invoke(connection.clj:11)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17905.invoke(threaded.clj:70)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2587)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:43)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:193)
        at toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)
        at toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17905.invoke(threaded.clj:70)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2587)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:43)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:193)
        at toucan2.pipeline$transduce_execute.invokeStatic(pipeline.clj:74)
        at toucan2.pipeline$transduce_execute.invoke(pipeline.clj:64)
        at clojure.lang.Var.invoke(Var.java:399)
        at toucan2.pipeline$transduce_compiled_query.invokeStatic(pipeline.clj:244)
        at toucan2.pipeline$transduce_compiled_query.invoke(pipeline.clj:240)
        at toucan2.pipeline$transduce_built_query.invokeStatic(pipeline.clj:252)
        at toucan2.pipeline$transduce_built_query.invoke(pipeline.clj:246)
        at toucan2.pipeline$transduce_query_primary_method_default.invokeStatic(pipeline.clj:272)
        at toucan2.pipeline$transduce_query_primary_method_default.invoke(pipeline.clj:269)
        at clojure.lang.AFn.applyToHelper(AFn.java:178)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17911.invoke(threaded.clj:79)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.doInvoke(core.clj:2589)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$apply.doInvoke(core.clj:662)
        at clojure.lang.RestFn.invoke(RestFn.java:533)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.doInvoke(threaded.clj:46)
        at clojure.lang.RestFn.applyTo(RestFn.java:151)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:482)
        at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___25175$with_transaction_STAR___25176.invoke(before_insert.clj:37)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at metabase.db.connection$do_transaction$thunk__32276.invoke(connection.clj:150)
        at metabase.db.connection$do_transaction.invokeStatic(connection.clj:162)
        at metabase.db.connection$do_transaction.invoke(connection.clj:146)
        at metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:199)
        at metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:172)
        at clojure.lang.AFn.applyToHelper(AFn.java:165)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:457)
        at clojure.core$partial$fn__5908.invoke(core.clj:2643)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17907.invoke(threaded.clj:71)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2588)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:44)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)
        at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)
        at clojure.lang.AFn.applyToHelper(AFn.java:165)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:457)
        at clojure.core$partial$fn__5908.invoke(core.clj:2643)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:195)
        at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___25175.invoke(before_insert.clj:36)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at toucan2.connection$bind_current_connectable_fn$fn__20813.invoke(connection.clj:104)
        at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invokeStatic(connection.clj:18)
        at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invoke(connection.clj:15)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17905.invoke(threaded.clj:70)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2587)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:43)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:193)
        at metabase.db.connection$do_with_connection_primary_method_default.invokeStatic(connection.clj:142)
        at metabase.db.connection$do_with_connection_primary_method_default.invoke(connection.clj:140)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17905.invoke(threaded.clj:70)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2587)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:43)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:193)
        at toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)
        at toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.combo.threaded$fn__17903$fn__17904$fn__17905.invoke(threaded.clj:70)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873$fn__17877.invoke(threaded.clj:23)
        at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
        at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
        at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
        at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
        at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at methodical.impl.combo.threaded$reducer_fn$fn__17873.invoke(threaded.clj:21)
        at clojure.core$comp$fn__5876.invoke(core.clj:2587)
        at methodical.impl.combo.threaded$combine_with_threader$fn__17883.invoke(threaded.clj:43)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
        at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:436)
        at clojure.core$partial$fn__5908.invoke(core.clj:2642)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
        at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:193)
        at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invokeStatic(before_insert.clj:36)
        at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invoke(before_insert.clj:31)
        at clojure.lang.AFn.applyToHelper(AFn.java:178)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
        at clojure.lang.RestFn.applyTo(RestFn.java:146)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:675)
        at clojure.core$apply.doInvoke(core.clj:662)
        at clojure.lang.RestFn.invoke(RestFn.java:533)
        at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
        at methodical.impl.standard$invoke_multifn.doInvoke(standard.clj:47)
        at clojure.lang.RestFn.invoke(RestFn.java:594)
        at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:199)
        at toucan2.pipeline$transduce_query_STAR_.invokeStatic(pipeline.clj:278)
        at toucan2.pipeline$transduce_query_STAR_.invoke(pipeline.clj:274)
        at toucan2.pipeline$transduce_with_model.invokeStatic(pipeline.clj:293)
        at toucan2.pipeline$transduce_with_model.invoke(pipeline.clj:280)
        at toucan2.pipeline$transduce_parsed.invokeStatic(pipeline.clj:309)
        at toucan2.pipeline$transduce_parsed.invoke(pipeline.clj:295)
        at toucan2.pipeline$transduce_unparsed.invokeStatic(pipeline.clj:317)
        at toucan2.pipeline$transduce_unparsed.invoke(pipeline.clj:311)
        at toucan2.pipeline$transduce_unparsed_with_default_rf.invokeStatic(pipeline.clj:374)
        at toucan2.pipeline$transduce_unparsed_with_default_rf.invoke(pipeline.clj:368)
        at toucan2.insert$insert_returning_instances_BANG_.invokeStatic(insert.clj:163)
        at toucan2.insert$insert_returning_instances_BANG_.doInvoke(insert.clj:154)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at metabase.models.serialization$fn__44643.invokeStatic(serialization.clj:612)
        at metabase.models.serialization$fn__44643.invoke(serialization.clj:610)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.models.serialization$default_load_one_BANG_.invokeStatic(serialization.clj:639)
        at metabase.models.serialization$default_load_one_BANG_.invoke(serialization.clj:632)
        at metabase.models.serialization$fn__44654.invokeStatic(serialization.clj:643)
        at metabase.models.serialization$fn__44654.invoke(serialization.clj:642)
        at metabase.models.field$fn__53286.invokeStatic(field.clj:459)
        at metabase.models.field$fn__53286.invoke(field.clj:458)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase_enterprise.serialization.v2.load$load_one_BANG_.invokeStatic(load.clj:67)
        ... 52 more
Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Referential integrity constraint violation: "FK_FIELD_PARENT_REF_FIELD_ID: PUBLIC.METABASE_FIELD FOREIGN KEY(PARENT_ID) REFERENCES PUBLIC.METABASE_FIELD(ID) (249)"; SQL statement:
INSERT INTO "METABASE_FIELD" ("DESCRIPTION", "DATABASE_TYPE", "SEMANTIC_TYPE", "TABLE_ID", "COERCION_STRATEGY", "NAME", "HAS_FIELD_VALUES", "SETTINGS", "CAVEATS", "FK_TARGET_FIELD_ID", "UPDATED_AT", "CUSTOM_POSITION", "EFFECTIVE
_TYPE", "ACTIVE", "NFC_PATH", "PARENT_ID", "DATABASE_IS_AUTO_INCREMENT", "JSON_UNFOLDING", "POSITION", "VISIBILITY_TYPE", "PREVIEW_DISPLAY", "DISPLAY_NAME", "DATABASE_POSITION", "DATABASE_REQUIRED", "CREATED_AT", "BASE_TYPE", "POINTS_OF_INTEREST") VALUES (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NULL, NULL, NOW(), ?, ?, TRUE, NULL, ?, FALSE, FALSE, ?, ?, TRUE, ?, ?, FALSE, ?, ?, NULL) [23506-214]
        ... 570 more
Command failed with exception: Failed to load into database for [#ordered/map ([:id "sample-db"] [:model "Database"]) #ordered/map ([:id "gs_7334"] [:model "Table"]) #ordered/map ([:id "level_2"] [:model "Field"])]
```

</details>

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "fr",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36 Edg/122.0.0.0",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Linux",
    "os.version": "5.15.133.1-microsoft-standard-WSL2",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-01-29",
      "tag": "v1.48.4",
      "hash": "62145b0"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```


### Severity

This is blocking our industrialization

### Additional context

_No response_

--- a/enterprise/backend/test/metabase_enterprise/serialization/v2/extract_test.clj
+++ b/enterprise/backend/test/metabase_enterprise/serialization/v2/extract_test.clj
@@ -1658,24 +1658,33 @@
                       (map serdes/path)
                       set)))))))))
 
-(deftest foreign-key-field-test
+(deftest field-references-test
   (mt/with-empty-h2-app-db
-    (ts/with-temp-dpc [Database   {db-id         :id}        {:name "My Database"}
-                       Table      {no-schema-id  :id}        {:name "Schemaless Table" :db_id db-id}
-                       Field      {some-field-id :id}        {:name "Some Field" :table_id no-schema-id}
-                       Table      {schema-id     :id}        {:name        "Schema'd Table"
+    (ts/with-temp-dpc [Database   {db-id          :id}        {:name "My Database"}
+                       Table      {no-schema-id   :id}        {:name "Schemaless Table" :db_id db-id}
+                       Field      {some-field-id  :id}        {:name "Some Field" :table_id no-schema-id}
+                       Table      {schema-id      :id}        {:name        "Schema'd Table"
                                                               :db_id       db-id
                                                               :schema      "PUBLIC"}
-                       Field      _                          {:name "Other Field" :table_id schema-id}
-                       Field      {fk-id         :id}        {:name     "Foreign Key"
+                       Field      {other-field-id :id}        {:name "Other Field" :table_id schema-id}
+                       Field      {fk-id          :id}        {:name     "Foreign Key"
                                                               :table_id schema-id
-                                                              :fk_target_field_id some-field-id}]
+                                                              :fk_target_field_id some-field-id}
+                       Field      {nested-id      :id}        {:name "Nested Field"
+                                                              :table_id schema-id
+                                                              :parent_id other-field-id}]
 
       (testing "fields that reference foreign keys are properly exported as Field references"
         (is (= ["My Database" nil "Schemaless Table" "Some Field"]
                (->> (t2/select-one Field :id fk-id)
                     (serdes/extract-one "Field" {})
-                    :fk_target_field_id)))))))
+                    :fk_target_field_id))))
+
+      (testing "Fields that reference parents are properly exported as Field references"
+        (is (= ["My Database" "PUBLIC" "Schema'd Table" "Other Field"]
+               (->> (t2/select-one Field :id nested-id)
+                    (serdes/extract-one "Field" {})
+                    :parent_id)))))))
 
 (deftest escape-report-test
   (mt/with-empty-h2-app-db

--- a/enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj
+++ b/enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj
@@ -126,7 +126,8 @@
           t1s        (atom nil)
           t2s        (atom nil)
           f1s        (atom nil)
-          f2s        (atom nil)]
+          f2s        (atom nil)
+          f3s        (atom nil)]
       (ts/with-dbs [source-db dest-db]
         (testing "serializing the two databases"
           (ts/with-db source-db
@@ -136,6 +137,7 @@
             (reset! t2s  (ts/create! Table    :name "posts" :db_id (:id @db2s))) ; Deliberately the same name!
             (reset! f1s  (ts/create! Field    :name "Target Field" :table_id (:id @t1s)))
             (reset! f2s  (ts/create! Field    :name "Foreign Key"  :table_id (:id @t2s) :fk_target_field_id (:id @f1s)))
+            (reset! f3s  (ts/create! Field    :name "Nested Field"   :table_id (:id @t1s) :parent_id (:id @f1s)))
             (reset! serialized (into [] (serdes.extract/extract {})))))
 
         (testing "serialization of databases is based on the :name"
@@ -152,10 +154,16 @@
         (testing "foreign key references are serialized as a field path"
           (is (= ["db1" nil "posts" "Target Field"]
                  (->> @serialized
-                      (filter #(-> % :serdes/meta last :model (= "Field")))
-                      (filter #(-> % :table_id (= ["db2" nil "posts"])))
-                      (keep :fk_target_field_id)
-                      first))))
+                      (u/seek #(and (-> % :serdes/meta last :model (= "Field"))
+                                    (-> % :name (= "Foreign Key"))))
+                      :fk_target_field_id))))
+
+        (testing "Parent field references are serialized as a field path"
+          (is (= ["db1" nil "posts" "Target Field"]
+                 (->> @serialized
+                      (u/seek #(and (-> % :serdes/meta last :model (= "Field"))
+                                    (-> % :name (= "Nested Field"))))
+                      :parent_id))))
 
         (testing "deserialization works properly, keeping the same-named tables apart"
           (ts/with-db dest-db
@@ -170,7 +178,11 @@
             (is (= #{(:id @db1d) (:id @db2d)}
                    (t2/select-fn-set :db_id Table :name "posts")))
             (is (t2/exists? Table :name "posts" :db_id (:id @db1d)))
-            (is (t2/exists? Table :name "posts" :db_id (:id @db2d)))))))))
+            (is (t2/exists? Table :name "posts" :db_id (:id @db2d)))
+            (is (= (t2/select-one-fn :id Field :name (:name @f1s))
+                   (t2/select-one-fn :fk_target_field_id Field :name (:name @f2s))))
+            (is (= (t2/select-one-fn :id Field :name (:name @f1s))
+                   (t2/select-one-fn :parent_id Field :name (:name @f3s))))))))))
 
 (deftest card-dataset-query-test
   ;; Card.dataset_query is a JSON-encoded MBQL query, which contain database, table, and field IDs - these need to be

--- a/src/metabase/models/field.clj
+++ b/src/metabase/models/field.clj
@@ -398,13 +398,15 @@
         (update :dimensions         extract-dimensions)
         (update :table_id           serdes/*export-table-fk*)
         (update :fk_target_field_id serdes/*export-field-fk*)
+        (update :parent_id          serdes/*export-field-fk*)
         (dissoc :fingerprint :last_analyzed :fingerprint_version))))
 
 (defmethod serdes/load-xform "Field"
   [field]
   (-> (serdes/load-xform-basics field)
       (update :table_id           serdes/*import-table-fk*)
-      (update :fk_target_field_id serdes/*import-field-fk*)))
+      (update :fk_target_field_id serdes/*import-field-fk*)
+      (update :parent_id          serdes/*import-field-fk*)))
 
 (defmethod serdes/load-find-local "Field"
   [path]

